solr/solrj/src/java/org/apache/solr/common/cloud/CollectionSetWatcher.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.common.cloud;
+
+import java.util.Set;
+
+/**
+ * Callback registered with {@link ZkStateReader#registerCollectionSetWatcher(CollectionSetWatcher)}
+ * and called whenever the cluster's collection set changes.
+ */
+public interface CollectionSetWatcher {
```

Comment:
What is a `CollectionSet`? It seems that this is watching collection states. Is `CollectionSetWatcher` the best name for this interface?

Reply:
There is likely a better name. I named it this because it is watching for the set of collections. `CollectionsWatcher` may be a better name, or `CollectionsListener`.